geometry/polyline2d.hpp

```diff
@@ -67,19 +67,25 @@ class PolylineT
 
   void Clear() { m_points.clear(); }
   void Add(Point<T> const & pt) { m_points.push_back(pt); }
-  void Swap(PolylineT & rhs)
+  void Append(Polyline const & poly)
   {
-    m_points.swap(rhs.m_points);
+    m_points.insert(m_points.end(), poly.m_points.cbegin(), poly.m_points.cend());
   }
 
-  size_t GetSize() const { return m_points.size(); }
+  void PopBack()
+  {
+    ASSERT(!m_points.empty(), ());
+    m_points.pop_back();
+  }
 
-  bool operator==(PolylineT<T> const & rhs) const { return m_points == rhs.m_points; }
+  void Swap(Polyline & rhs) { m_points.swap(rhs.m_points); }
+  size_t GetSize() const { return m_points.size(); }
+  bool operator==(Polyline const & rhs) const { return m_points == rhs.m_points; }
 
-  typedef vector<Point<T> > TContainer;
-  typedef typename TContainer::const_iterator TIter;
-  TIter Begin() const { return m_points.begin(); }
-  TIter End() const { return m_points.end(); }
+  using Container = vector<Point<T>>;
```

Comment:
Typedefs should go first: https://google.github.io/styleguide/cppguide.html#Declaration_Order